src/python/bindings.cpp

```diff
@@ -4,17 +4,36 @@
 //
 
 #include "cliffordsynthesis/CliffordSynthesizer.hpp"
+#include "cliffordsynthesis/GateSet.hpp"
 #include "exact/ExactMapper.hpp"
 #include "heuristic/HeuristicMapper.hpp"
 #include "nlohmann/json.hpp"
+#include "operations/OpType.hpp"
 #include "pybind11/pybind11.h"
 #include "pybind11/stl.h"
 #include "pybind11_json/pybind11_json.hpp"
 #include "python/qiskit/QuantumCircuit.hpp"
 
+#include <vector>
+
 namespace py = pybind11;
 using namespace pybind11::literals;
 
+cs::GateSet loadGateSet(const py::object& circ) {
+  cs::GateSet gs{};
+  if (py::isinstance<py::list>(circ)) {
+    auto l = circ.cast<py::list>();
+    if (l.size() > 0) {
+      for (auto& gate : l) {
+        if (py::isinstance<py::str>(gate)) {
+          gs.emplace_back(qc::opTypeFromString(gate.cast<std::string>()));
+        }
+      }
+    }
+  }
+  return gs;
+}
```

Comment:
if this only works with lists, then the function signature should reflect that. It would be nice to explore whether pybind even allows you to request a list of strings. This the function as it is could be called with quite some garbage.
```suggestion
cs::GateSet loadGateSet(const py::list& gateSet) {
  cs::GateSet gs{};
  if (gateSet.size() > 0) {
    for (auto& gate : gateSet) {
      if (py::isinstance<py::str>(gate)) {
        gs.emplace_back(qc::opTypeFromString(gate.cast<std::string>()));
      }
    }
  }
  return gs;
}
```